pkg/storage/provider_no_send_test.go

```diff
@@ -3,162 +3,378 @@ package storage_test
-	// given:
-	const inputSatoshis = 99904
+		// and:
+		given.Faucet(activeStorage, testusers.Alice).TopUp(inputSatoshis) // Intentionally, create only one UTXO
 
-	fixture := testabilities.NewBuildNoSendTransactionFixture(t, inputSatoshis)
-	defer fixture.Cleanup()
-
-	result := fixture.BuildNoSendTransaction()
-
-	// Step 3. Create an Action with NoSendChange using the result of the previous Create Action. Call CreateAction with IsNewTx set to true, including the two previous transactions in SendWithSlice.
-	createActionArgs := fixtures.DefaultValidCreateActionArgs(func(args *wdk.ValidCreateActionArgs) {
-		args.Outputs[0].Satoshis = 1
-		args.IsNoSend = true
-		args.Options.NoSend = to.Ptr(primitives.BooleanDefaultFalse(true))
-		args.Options.NoSendChange = slices.Map(result.SecondCreateActionResult.NoSendChangeOutputVouts, func(vout int) wdk.OutPoint {
-			return wdk.OutPoint{
-				TxID: result.SecondTxID,
-				Vout: uint32(vout),
-			}
+		// when:
+		var noSendChangeOutpoints []wdk.OutPoint
+		for i := range noSendChainCount {
+			t.Logf("Creating NoSend tx #%d", i+1)
+			noSendChangeOutpoints = givenNoSend.CreateAndProcessNoSendAction(noSendChangeOutpoints)
+		}
+
+		// and:
+		// Call processAction using sendWith and IsNewTx set to false, including the two previous transactions in SendWithSlice.
+		sendWithProcessAction := givenNoSend.ProcessAction(wdk.ProcessActionArgs{
+			IsNewTx:    false,
+			IsNoSend:   false,
+			SendWith:   givenNoSend.NoSendTxsHexStrings(),
+			IsSendWith: true,
 		})
-		args.IsNewTx = true
-		args.IsSendWith = true
-		args.Options.SendWith = []primitives.HexString{
-			primitives.HexString(result.FirstTxID),
-			primitives.HexString(result.SecondTxID),
+
+		// then:
+		testabilities.NotDelayedResultsAsserter(sendWithProcessAction.NotDelayedResults).
+			ContainsTxsWithStatus(t, wdk.ReviewActionResultStatusSuccess, givenNoSend.NoSendTxs()...)
+
+		testabilities.SendWithResultsAsserter(sendWithProcessAction.SendWithResults).
+			ContainsTxsWithStatus(t, wdk.SendWithResultStatusUnproven, givenNoSend.NoSendTxs()...)
+
+		testabilities.
+			ThenDBState(t, activeStorage).
+			HasUserTransactionsByTxIDsWithStatus(testusers.Alice, wdk.TxStatusUnproven, givenNoSend.NoSendTxs()...)
+
+		testabilities.ThenFunds(t, testusers.Alice, activeStorage).
+			ShouldNotBeAbleToReserveSatoshis(1) // All funds are tied up in the long NoSend chain
+	})
+
+	t.Run("two no-send txs, two initial UTXOs (one used for txs chain), not all noSendChange used, no newTx when sendWith", func(t *testing.T) {
+		// NOTE: In this case, the balance after all should be made of one initial UTXO + not used noSendChanges + last no-send tx change
+		// given:
+		const initialUTXOSats = 10000
+		const noSendChainCount = 2
+
+		given, cleanup := testabilities.Given(t)
+		defer cleanup()
+
+		activeStorage := given.Provider().GORM()
+		givenNoSend := testabilities.GivenNoSend(t, given, activeStorage, testusers.Alice)
+
+		// and:
```

Comment:
maybe add more details to those comments, prepare dedicated methods in fixture (Given)


Here for example could be
```suggestion
		// and: user has 2 UTXO each one higher then the amount spend in all no send transactions
```

Reply:
After refactor the code should be self explanatory enough 🤞